### PR TITLE
[RHCLOUD-21463] Added README.md and meta.json to the zipped payload

### PR DESCRIPTION
**Pull request by justinorringer**
### Changes
- Adds a readme and json file to the payload **on upload to s3.**
  - There are example files: `example_export_download.zip` and `example_meta.json`
- Added a GetFilters() helper to the models package.

### Steps to make and download a payload
- First create an export (the makefile uses the `example_export_request.json` file).
  - `make make sample-request-create-export`
- The example file has 2 sources now, so we need to use the internal upload endpoint to add the content from each source. Copy the export payload id from the above command and add it to the Makefile's EXPORT_ID variable. Also, do the same for one of the export resources, EXPORT_RESOURCE var.
- Then upload the `example-export-upload.zip` file as the requested resource.
  - `make sample-request-internal-upload`
- Do the same for the second source.
- Now, download the payload:
  - `make sample-request-export-download`
- and unzip it, which will yield the resources, the meta.json, and the Readme.md.

### CODE REVIEW

**Review by CodyWMitchell — DISMISSED** [review]
Looks great! 

In the future, we need to add info about error-ed export resources.

**Review by CodyWMitchell — DISMISSED** [review]
Looks good to me! :+1:

File: s3/compress.go [hunk 1]
```diff
@@ -47,17 +48,18 @@ func GetObjects(c context.Context, api S3ListObjectsAPI, input *s3.ListObjectsV2
 	return api.ListObjectsV2(c, input)
 }
 
-func (c *Compressor) zipExport(ctx context.Context, t time.Time, prefix, filename, s3key string) error {
+func (c *Compressor) zipExport(ctx context.Context, t time.Time, prefix, filename, s3key string, meta ExportMeta, sources []*models.Source) error {
```
Comment (dehort) [inline s3/compress.go:51]: I don't think the `t time.Time` parameter is required.

File: s3/compress.go [hunk 2]
```diff
@@ -134,7 +191,19 @@ func (c *Compressor) Compress(ctx context.Context, m *models.ExportPayload) (tim
...
+	if err != nil {
+		return t, filename, s3key, fmt.Errorf("failed to get sources: %w", err)
+	}
+
+	meta := ExportMeta{
+		ExportBy:    m.User.Username,
+		ExportDate:  m.CreatedAt.Format(time.RFC3339),
+		ExportOrgID: m.User.OrganizationID,
+		HelpString:  helpString,
+	}
```
Comment (dehort) [inline s3/compress.go:204]: Should building this object be the responsiblity of the zipExport() method?  Just something to think about.

**Review by dehort — DISMISSED** [review]
LGTM!  I left a couple of comments.

In the future, we might want to look into using golden files for the README tests.

I wonder if it might also simplify the code a bit if we used the `text/template` package to build the readme.

I don't want to block this PR.  Just a couple of things to think about for the future.

File: s3/template.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+type ExportFileMeta struct {
+	Filename    string `json:"filename"`
+	Application string `json:"application"`
+	Resource    string `json:"resource"`
+	// Filters are a key-value pair of the filters used to create the export
+	Filters map[string]string `json:"filters"`
+}
+
+const (
+	helpString = `Contained in this archive are your requested resources. If you need help or have any questions, please contact our team on slack @crc-pipeline-team.`
```
[the remark this answers is not in the document]
Comment (justinorringer) [reply s3/template.go:33]: Sure! I copied that from the Changelog readme, sounds good!

File: s3/compress.go (hunk 2 above)
Comment (justinorringer) [reply s3/compress.go:204]: I went back and forth on that one. I ended up doing it this way, so it wouldnt pass down the ExportPayload model.

Comment (dehort) [reply s3/compress.go:204]: I think its fine as it is.

**Review by CodyWMitchell — APPROVED** [review]
LGTM! :+1:

**Pull request merged**